api/users/views.py

```diff
@@ -574,3 +577,47 @@ def create(self, request, *args, **kwargs):
         user.requested_deactivation = True
         user.save()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class UserChangePassword(JSONAPIBaseView, generics.CreateAPIView, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    required_read_scopes = [CoreScopes.NULL]
+    required_write_scopes = [CoreScopes.USER_SETTINGS_WRITE]
+
+    view_category = 'users'
+    view_name = 'user-password'
+
+    serializer_class = UserChangePasswordSerializer
+    throttle_classes = (SendEmailThrottle, )
+
+    def create(self, request, *args, **kwargs):
```

Comment:
I get why you have to override the create method here to return the correct response, but pretty sure all the validation/actual setting of the information should happen in the `UserChangePasswordSerializer` - I'd move that logic there.